packages/peregrine/lib/talons/RootComponents/Product/useProduct.js

```diff
@@ -29,14 +34,23 @@ export const useProduct = props => {
     });
 
     const product = useMemo(() => {
+        if (!data) {
+            // The product isn't in the cache and we don't have a response from GraphQL yet.
+            return null;
+        }
+
+        const supportedProducts = data.products.items.filter(
```

Comment:
To future proof this fix (in a distant future where we support grouped products), we can instead just search for the `url_key` we intended to query for. `const product = data.product.items.find(item => item.url_key === urlKey)`